neuraxle/steps/caching.py

```diff
@@ -83,15 +72,17 @@ def handle_fit_transform(self, data_container: DataContainer) -> ('BaseStep', Da
 
         return self, data_container
 
-    def handle_transform(self, data_container: DataContainer) -> DataContainer:
+    def handle_transform(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
         """
         Transform data container.
 
+        :param context: execution context
         :param data_container: the data container to transform
         :type data_container: DataContainer
 
         :return: transformed data container
         """
+        self.create_checkpoint_path(context.get_path())
```

Comment:
same comment, this line feels weird because this should be the responsibility of the data checkpoints methods and of the steps' save methods.